src/models/clap.py

```diff
@@ -0,0 +1,38 @@
+import torch
+from .audio import BaseAudioEncoder
+from .text import BaseTextEncoder
+from typing import Union, List, Dict
+import numpy as np
+
+
+class BaseClap(torch.nn.Module):
+    def __init__(
+            self,
+            audio_encoder: BaseAudioEncoder,
+            text_encoder: BaseTextEncoder
+    ):
+        super().__init__()
+        self.audio_encoder = audio_encoder
+        self.text_encoder = text_encoder
+        self.loss_fn = torch.nn.CrossEntropyLoss()
+
+    def forward(self, texts: Union[List[str], str], audios: Union[List[np.array], np.array]) -> Dict[str, torch.Tensor]:
+        """
+        :param texts: texts for the input
+        :param audios: audios for the
+        :return: dictionary with cosine distance between embeddings
+            if len(texts) == len(audios) also returns the loss
```

Comment:
better have loss always, but it can be None sometimes. Even better torch.NaN

Reply:
Hmm, maybe you are right, I will do it

Reply:
Like, KeyError is worse then NoneError